Sources/Basics/Concurrency/AsyncBox.swift

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=5.5.2)
+
+import struct Foundation.Date
+
+/// A `Sendable` storage that allows access from concurrently running tasks in an `async` closure.
+public actor AsyncBox<Value: Sendable> {
```

Comment:
curious, why is this even needed?

Reply:
In `HTTPClient` tests we need a counter for checking how many times a closure was invoked and we need to assert on that counter after the closure has run. Previously this was done with a lock. Now with Swift Concurrency `async` closures can't capture an arbitrary variably, but they can capture references to actors.

Reply:
Ah, I remember the details now. It's a direct replacement for `ThreadSafeBox`.

Reply:
okay - maybe `SendableBox` then?